### PR TITLE
fix assertion against layer attributes

### PR DESCRIPTION
**Pull request by harts-boundless**
This assertion was always succeeding because the value of the Python expression `('xsd:dateTime' or ...)` is always `'xsd:dateTime'`, which is truthy, so that `assertTrue` passes regardless of what is happening. One could guess this wouldn't be intended, but the comment `make sure we have at least one dateTime attribute` makes the original intention very clear.

To complicate matters, it looks like there were several tests which should have been failing before, and now are failing because this assertion has been fixed, which will cause any future PRs to fail the tests. It would help me if someone more familiar with the date requirement could fix these test failures so that the test suite can be used to validate future PRs that are in the works.

Please don't hesitate to tell me if there are any adjustments you would like to have made to this PR.

### CODE REVIEW

**Comment by sarasafavi** [comment]
TL;DR: this check (to "make sure we have at least one dateTime attribute") can be removed.

After investigation, it looks like this check was from the original (Mapstory-era) test code, where dateTimes were required. The failures that occur now (with this PR's changes) are accurate; the test data it fails on do not have date/datetime fields.

**Comment by harts-boundless** [comment]
@garnertb thanks, again please let me know if you all have any feedback or input on these PRs.

@sarasafavi thank you for your input, that sounds like the right way to go.
